### PR TITLE
Tweak config and exception

### PR DESCRIPTION
**Pull request by Peanuuutz**
[here](https://github.com/akuleshov7/ktoml/issues/49#issuecomment-1009497563)

### CODE REVIEW

File: ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/exceptions/TomlSerializationExceptions.kt
Comment (akuleshov7) [inline ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/exceptions/TomlSerializationExceptions.kt:11]: If you have finally renamed it - may be make all exception's naming consistent? 
Let's select one of two variants:
1) All of them are with 'Toml' prefix
2) All of them are without a `Toml` prefix and only the base class `TomlDecodingException` contains it

**Comment by akuleshov7** [comment]
Looks good, but it will completely break all users in the community 😭 😭 😭

**Comment by Peanuuutz** [comment]
> Looks good, but it will completely break all users in the community 😭 😭 😭 

I have that in mind, so this problem could be solved later. Just put an @Deprecated to inform users about this change. I'll rollback soon and commit a new one like I said.

File: ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/exceptions/TomlSerializationExceptions.kt
Comment (Peanuuutz) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/exceptions/TomlSerializationExceptions.kt]: > All of them are without a `Toml` prefix and only the base class `TomlDecodingException` contains it

I prefer this, and I imagine something like `TomlEncodingException` for encoding. I'd like to make both of these as `public` for users to manipulate, while actual exception types keep `internal`.

Comment (Peanuuutz) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/exceptions/TomlSerializationExceptions.kt]: A further question: If there is only one file in exception folder, I think to put it out along with `Toml` and other stuff may be better, as that folder seems redundant.

Comment (akuleshov7) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/exceptions/TomlSerializationExceptions.kt]: >>> I prefer this, and I imagine something like TomlEncodingException for encoding
totally agree: `TomlEncodingException`/`TomlDecodingException` sounds very good

Comment (akuleshov7) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/exceptions/TomlSerializationExceptions.kt]: > A further question: If there is only one file in exceptions folder, I think to put it out along with Toml and other stuff may be better, as that folder seems redundant.

Good idea, the directory is redundant. I just automatically used some useless Java practices in Kotlin 😃

Comment (Peanuuutz) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/exceptions/TomlSerializationExceptions.kt]: > Hm, but that will change package name. Instead of ktoml.exceptions.BlaBlaException - you will have ktoml.BlaBlaException. May be better to have it as is...

Then I'll have `TomlEncodingException` in one file, and `TomlDecodingException` another one.

**Comment by Peanuuutz** [comment]
Done.
- Marked `KtomlConf` as `@Deprecated`, and created `TomlConfig` in advance(plan to migrate on 0.4.0).
- Tweak exception classes.
- Changed `field` to `property` because the latter is preferred in Kotlin.

**Review by akuleshov7 — APPROVED** [review]
lgtm, let's merge it after #90
and please fix diktat and detekt issues
If you don't mind - I will add you to this repo as a maintainer, so I won't need to approve CI/CD  actions each time

image

**Comment by Peanuuutz** [comment]
diKTat errors this: `public value class ...` should be `value public class ...`, which makes no sense to me. 😶

**Comment by akuleshov7** [comment]
> diKTat errors this: `public value class ...` should be `value public class ...`, which makes no sense to me. 😶

created https://github.com/analysis-dev/diktat/issues/1170 and assigned on @Cheshiriks. I guess it should be fixed in the nearest future. 

@Peanuuutz could you please suppress this case with a @Suppress("WRONG_MULTIPLE_MODIFIERS_ORDER")?

**Comment by akuleshov7** [comment]
If possible - could you please add the section `Acknowledgement` to the readme in this PR and mention something like `Thanks to` NightEule5 Peanuuutz petertrr Olivki and edrd-f  there?

**Comment by Peanuuutz** [comment]
I'm not good at paperwork so feel free to edit. :P

**Comment by akuleshov7** [comment]
> I'm not good at paperwork so feel free to edit. :P

ok, sure 😄

**Comment by Peanuuutz** [comment]
Sure 💪

**Comment by akuleshov7** [comment]
> Sure 💪

oh, it's in fork. Never mind, I will open a new PR :)

**Review by Peanuuutz — COMMENTED** [review]
Reviewed again. Ready to merge.

**Comment by akuleshov7** [comment]
> Reviewed again. Ready to merge.

Merged, but as you are a maintainer of this repo, can merge by yourself next time. But please use squash-merge

**Pull request merged**
